.github/workflows/documentation.yml

```diff
@@ -0,0 +1,65 @@
+name: Documentation
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+
+  doc:
+    name: Build and deploy documentation
+    runs-on: ubuntu-latest
+    env:
+      GO_VERSION: '1.20'
+      NODE_VERSION: '20.x'
+      CGO_ENABLED: 0
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go ${{ env.GO_VERSION }}
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Cache Go modules
+        uses: actions/cache@v3
+        with:
+          # In order:
+          # * Module download cache
+          # * Build cache (Linux)
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+          key: docs-${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            docs-${{ runner.os }}-go-
+
+      - run: go mod download
+
+      - name: Use Node.js ${{ env.NODE_VERSION }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: npm
+          cache-dependency-path: docs/package-lock.json
+
+      - run:  npm install --legacy-peer-deps
+        working-directory: ./docs
+
+      - name: Build Documentation
+        run: npm run build
+        working-directory: ./docs
+
+      - name: Deploy to GitHub Pages
+        uses: crazy-max/ghaction-github-pages@v3
```

Comment:
Not important but any reason you went for this implementation? I only have experience from one action deploying to GitHub pages and that's with [peaceiris/actions-gh-pages](https://github.com/peaceiris/actions-gh-pages). It seems to be way more active with 10x stars.

Also there seems to be an official action [actions/deploy-pages](https://github.com/actions/deploy-pages) but it doesn't seem as easy to work with.

Reply:
No particular reason other than it's the only one I've ever used.